### PR TITLE
Watersmart 604

### PR DESCRIPTION
**Pull request by mbucknell**
Refactored the service calls to retrieve measured streamflow and modeled streamflow into there own modules. The retrieval results can be customized by passing in conversion functions for the date and for the value. 

Added the modeled streamflow data to the water budget accumulated pages if it is available for the chosen huc12.

Added the modeled huc layer to the Water Budget map select page.

### CODE REVIEW

File: nwc/src/main/webapp/js/utils/dataSeriesStore.js [hunk 1]
```diff
@@ -1,461 +1,547 @@
-var NWC = NWC || {};
```
Comment (mbucknell) [inline nwc/src/main/webapp/js/utils/dataSeriesStore.js]: Looks like the previous file had whitespace issues. Added the ability to add the modeledStreamflow data and adapted the update* functions to handle the situation where the measured streamflow is missing but the modeled streamflow is available. This module could use some serious refactoring.

File: nwc/src/main/webapp/js/view/HucInsetMapView.js [hunk 2]
```diff
@@ -68,7 +77,16 @@ NWC.view = NWC.view || {};
 			this.map = NWC.util.mapUtils.createMap([baseLayer], mapControls);
 
 			// Load vector layers
-			this.featureLoadedPromise = $.when(hucLoadedDeferred, achucLoadedDeferred, gageLoadedDeferred);
+			this.featureLoadedPromise = $.Deferred();
+			$.when(hucLoadedDeferred, achucLoadedDeferred, gageLoadedDeferred, modeledHucLoadedDeferred).done(function(d1, d2, d3, d4) {
+				var result = {};
```
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/HucInsetMapView.js]: Doing this so any args passed back from the data loading deferred is passed on to the featureLoadedPromise. This way I can let the caller of this.featureLoadedPromise know whether modeled streamflow was retrieved successfully.

File: nwc/src/main/webapp/js/view/StreamflowStatsGageDataView.js [hunk 3]
```diff
@@ -94,14 +94,6 @@ NWC.view.StreamflowStatsGageDataView = NWC.view.BaseView.extend({
 	render : function() {
 		NWC.view.BaseView.prototype.render.apply(this, arguments);
 		this.map.render(this.insetMapDiv);
-		this.streamflowPlotViewLeft = new NWC.view.StreamflowPlotView({
```
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/StreamflowStatsGageDataView.js]: Changes in this module in addition to calling fetchMeasuredStreamFlow included changes to eliminate unnecessary view level promise attributes that aren't needed. This hopefully simplifies the flow.

File: nwc/src/main/webapp/js/view/StreamflowStatsHucDataView.js [hunk 4]
```diff
@@ -27,16 +27,7 @@ NWC.view = NWC.view || {};
 
 		render : function() {
 			NWC.view.BaseView.prototype.render.apply(this, arguments);
-
```
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/StreamflowStatsHucDataView.js]: Similar changes to what was made in StreamflowStatsGageDataView.

File: nwc/src/main/webapp/js/view/WaterbudgetPlotView.js [hunk 5]
```diff
@@ -42,11 +43,11 @@ NWC.view = NWC.view || {};
 			this.compare = options.compare ? options.compare : false;
 
 			NWC.view.BaseView.prototype.initialize.apply(this, arguments);
-			this.getHucDataPromise = this.getPlotData(options.hucId, this.gageId).done(function() {
+			this.getPlotData(options.hucId, this.gageId, options.hasModeledStreamflow).done(function() {
 				self.$el.find('.download-btn-container button').prop('disabled', false);
+				self.plotData.bind(self)();
```
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/WaterbudgetPlotView.js]: Calling plotData from the done on getPlotData rather then putting the promise property on the view and waiting for that to resolve in plotData.

**Comment by instantsoup** [comment]
Nice work!

**Pull request merged**